textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/dictionarymatcher/DictionaryMatcherTest.java

```diff
@@ -13,153 +14,243 @@
-            Assert.assertTrue(contains);
-            numTuples++;
-        }
-        Assert.assertEquals(4, numTuples);
-
-    }
-
-    /**
-     * Scenario S2:verifies GetNextTuple of DictionaryMatcher and single word
-     * queries
-     */
-
-    @Test
-    public void testGetNextTuple() throws Exception {
-
-        ArrayList<String> names = new ArrayList<String>(Arrays.asList("bruce","cena","lin","george"));
-        IDictionary dictionary = new Dictionary(names);
-        ISourceOperator sourceOperator = new ScanBasedSourceOperator(dataReader);
-        List<ITuple> data = TestConstants.getSamplePeopleTuples();
-        List<Attribute> attributes = TestConstants.ATTRIBUTES_PEOPLE;
-        dictionaryMatcher = new DictionaryMatcher(dictionary, sourceOperator);
-        dictionaryMatcher.open();
-        ITuple iTuple;
-        int numTuples=0;
-        while ((iTuple = dictionaryMatcher.getNextTuple()) != null) {
-            
-            boolean contains = TestUtils.checkSpan(data, iTuple, attributes);
-            Assert.assertTrue(contains);
-            numTuples++;
+	private DictionaryMatcher dictionaryMatcher;
+	private LuceneDataStore dataStore;
+	private IDataWriter dataWriter;
+	private IDataReader dataReader;
+	private Analyzer analyzer;
+	private Query query;
+
+	@Before
+	public void setUp() throws Exception {
+
+		dataStore = new LuceneDataStore(LuceneConstants.INDEX_DIR, TestConstants.SCHEMA_PEOPLE);
+		analyzer = new StandardAnalyzer();
+		dataWriter = new LuceneDataWriter(dataStore, analyzer);
+		QueryParser queryParser = new QueryParser(TestConstants.ATTRIBUTES_PEOPLE[0].getFieldName(), analyzer);
+		query = queryParser.parse(LuceneConstants.SCAN_QUERY);
+		dataReader = new LuceneDataReader(dataStore, query);
+		dataWriter.clearData();
+		dataWriter.writeData(TestConstants.getSamplePeopleTuples());
+
+	}
+
+	@After
+	public void cleanUp() throws Exception {
+		dataWriter.clearData();
+	}
+
+	public List<ITuple> getQueryResults(IDictionary dictionary, ISourceOperator sourceOperator, List<Attribute> attributes) throws Exception {
+
+		dictionaryMatcher = new DictionaryMatcher(dictionary, sourceOperator, attributes);
+		dictionaryMatcher.open();
+		ITuple nextTuple = null;
+		List<ITuple> results = new ArrayList<ITuple>();
+		while ((nextTuple = dictionaryMatcher.getNextTuple()) != null) {
+			results.add(nextTuple);
+		}
+		dictionaryMatcher.close();
+		return results;
+	}
+
+	/**
+	 * Scenario S1:verifies GetNextTuple of Dictionary
+	 */
+
+	@Test
+	public void testGetNextOfDictionaryItem() throws Exception {
+
+		ArrayList<String> names = new ArrayList<String>(Arrays.asList("george", "lee"));
+		IDictionary dictionary = new Dictionary(names);
+		int numTuples = 0;
+		String dictionaryItem;
+		while ((dictionaryItem = dictionary.getNextValue()) != null) {
+			boolean contains = TestUtils.contains(names, dictionaryItem);
+			Assert.assertTrue(contains);
+			numTuples++;
+		}
+		Assert.assertEquals(2, numTuples);
```

Comment:
Why didn't we use the "containsAllResults()" helper function for this test case?